templates/channels/channel_configuration.haml

```diff
@@ -510,6 +510,33 @@
       %code= ip_address
 
 
+  -elif object.channel_type == 'FCM'
+    %h4
+      -trans "To use your Firebase Cloud Messaging channel you'll have to POST to the following URLs with the parameters below."
+
+    .clearfix
+    %hr
+
+    %h4
+      -trans "Contact Register"
+
+    %p
+      -trans "To register contacts, POST to the following URL with the parameters urn, fcm_token and optionally name."
+
+    %code
+      https://{{ domain }}{% url 'handlers.fcm_handler' 'register' object.uuid %}
+
+    %h4
+      -trans "Receive URL"
+
+    %p
+      -trans "To handle incoming messages, POST to the following URL with the parameters from, msg and fcm_token."
+
+    %code
+      https://{{ domain }}{% url 'handlers.fcm_handler' 'receive' object.uuid %}
+
+    %hr
+
```

Comment:
Ah, ok, after doing some more reading I understand now that we aren't adding upstream messaging here (device to RapidPro via Firebase) only downstream. MO messages are delivered directly via the HTTP handler above and hit directly by the Android app, is that right?